governance/proposals/008-uniswap-v2-remaining-liquidity-tokens.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Migrating Unlock DAO's remaining UDT/WETH Liquidity Position
+ * from Uniswap v2 pool to the Uniswap v3 pool
+ */
+const { ethers } = require('hardhat')
+const { getTokenInfo } = require('../helpers/uniswap')
+
+// const daoAddress = '0x440d9D4E66d39bb28FB58729Cb4D3ead2A595591'
+const timelockAddress = '0x17EEDFb0a6E6e06E95B3A1F928dc4024240BC76B'
+
+// remaining lp tokens
+const percentageToMigrate = 100
+const multisigAddress = '0xc3F2bcBE6fB42faC83dd8a42893C6c0B3809e070'
+const udtWethV2poolAddress = '0x9cA8AEf2372c705d6848fddA3C1267a7F51267C1'
+
+const poolV2Abi = [
+  'function getReserves() external view returns (uint112 reserve0, uint112 reserve1, uint32 blockTimestampLast)',
+  'function token0() external view returns (address)',
+  'function token1() external view returns (address)',
+  'function balanceOf(address) external view returns (uint256)',
+  'function transfer(address,uint256) external view',
+  'function approve(address,uint256) external',
+  'function totalSupply() external view returns (uint256)',
+  'function allowance(address, address) external view returns (uint256)',
+]
+
+module.exports = async () => {
+  console.log(
+    `Migrate ${percentageToMigrate}% of liquidity from v2 to multisig`
+  )
+
+  // parse call data for function call
+  const poolV2 = await ethers.getContractAt(poolV2Abi, udtWethV2poolAddress)
+  const poolV2Address = await poolV2.getAddress()
+
+  // pool v2 info
+  const [token0, token1, [reserve0, reserve1], totalSupply] = await Promise.all(
+    [
+      poolV2.token0(),
+      poolV2.token1(),
+      poolV2.getReserves(),
+      poolV2.totalSupply(),
+    ]
+  )
+
+  const [{ symbol: symbol0 }, { symbol: symbol1 }] = await Promise.all([
+    await getTokenInfo(token0),
+    await getTokenInfo(token1),
+  ])
+
+  // how many ERC20 owned by the timelock
+  const liquidity = await poolV2.balanceOf(timelockAddress)
+
+  // calculate the share of the entire liquidity for each token
+  // to determine the amount of tokens owned by the timelock in the pool
+  const lp0 = (liquidity * reserve0) / totalSupply
```

Comment:
`.mul()` instead of `*` and `.div()` instead of `/` ?

Reply:
yes ethers v6 has added support for native js BigInt